src/hub/Hub.sol

```diff
@@ -619,4 +641,8 @@ contract Hub is Multicall, Auth, Recoverable, IHub, IHubRequestManagerCallback,
     function _isManager(PoolId poolId) internal view {
         require(hubRegistry.manager(poolId, msg.sender), IHub.NotManager());
     }
+
+    function _payment() internal view returns (uint256 value) {
+        return gateway.isBatching() ? 0 : msg.value;
+    }
```

Comment:
We can only pay for messages if we're not in a multicall, if not the first message in the multicall will take all `msg.value` making rest methods to revert:

Related to this: https://kflabs.slack.com/archives/C07PG2EUR9C/p1759183383159389